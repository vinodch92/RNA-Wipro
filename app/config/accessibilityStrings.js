export default {
  AUTHOR_NAME_TEXT: 'author-name-text',
};
