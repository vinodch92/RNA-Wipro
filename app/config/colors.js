export default {
  black: '#000000',
  white: '#FFFFFF',
  appTheme: '#34424A',
};
