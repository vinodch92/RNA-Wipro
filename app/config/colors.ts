export default {
  black: '#000000',
  gray: '#9B9B9B',
  white: '#FFFFFF',
  appTheme: '#1A1924',
};
