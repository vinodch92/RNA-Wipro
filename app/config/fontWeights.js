export default {
  FONT_WEIGHT_THIN: '100', // Thin
  FONT_WEIGHT_ULTRALIGHT: '200', // Ultra Light
  FONT_WEIGHT_LIGHT: '300', // Light
  FONT_WEIGHT_REGULAR: '400', // Regular
  FONT_WEIGHT_MEDIUM: '500', // Medium
  FONT_WEIGHT_SEMIBOLD: '600', // Semibold
  FONT_WEIGHT_BOLD: '700', // Bold
  FONT_WEIGHT_HEAVY: '800', // Heavy
  FONT_WEIGHT_BLACK: '900', // Black
};
