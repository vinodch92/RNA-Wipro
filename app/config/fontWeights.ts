type FontWeight =
  | 'normal'
  | 'bold'
  | '100'
  | '200'
  | '300'
  | '400'
  | '500'
  | '600'
  | '700'
  | '800'
  | '900';

interface FontWeights {
  [key: string]: FontWeight;
}

const FONT_WEIGHT_THIN = '100'; // Thin
const FONT_WEIGHT_ULTRALIGHT = '200'; // Ultra Light
const FONT_WEIGHT_LIGHT = '300'; // Light
const FONT_WEIGHT_REGULAR = '400'; // Regular
const FONT_WEIGHT_MEDIUM = '500'; // Medium
const FONT_WEIGHT_SEMIBOLD = '600'; // Semibold
const FONT_WEIGHT_BOLD = '700'; // Bold
const FONT_WEIGHT_HEAVY = '800'; // Heavy
const FONT_WEIGHT_BLACK = '900'; // Black

const FONT_WEIGHTS: FontWeights = {
  FONT_WEIGHT_THIN,
  FONT_WEIGHT_ULTRALIGHT,
  FONT_WEIGHT_LIGHT,
  FONT_WEIGHT_REGULAR,
  FONT_WEIGHT_MEDIUM,
  FONT_WEIGHT_SEMIBOLD,
  FONT_WEIGHT_BOLD,
  FONT_WEIGHT_HEAVY,
  FONT_WEIGHT_BLACK,
};

export default FONT_WEIGHTS;
