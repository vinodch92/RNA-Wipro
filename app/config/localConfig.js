export default {
  storybookEnabled: false,
  imageWidth: 200,
  imageHeight: 300,
  imagesDataBaseUrl: 'https://picsum.photos',
};
