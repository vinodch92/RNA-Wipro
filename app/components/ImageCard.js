import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import config from '../config';

/**
 * This functional component is ImageCard and useful to show images with author name for image slider.
 * @param {*} params : params consists of id of image, authorName, containerStyle object for customizing
 * main container, textStyle object for customizing authorName text style, imageWrapperStyle object for
 * customizing style of wrapper of image, textNumLines to specify the number of lines for Text element.
 */
const ImageCard = ({
  id = 0,
  authorName,
  containerStyle,
  textStyle,
  textNumLines,
  imageWrapperStyle,
}) => {
  const imageSource = `${config.imagesDataBaseUrl}/${config.imageWidth}/${config.imageHeight}?image=${id}`;

  return (
    <View style={[styles.container, containerStyle]}>
      {authorName ? (
        <Text style={[styles.text, textStyle]} numberOfLines={textNumLines}>
          {authorName}
        </Text>
      ) : null}
      <View style={[styles.imageWrapper, imageWrapperStyle]}>
        <Image style={styles.image} source={{ uri: imageSource }} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: { alignItems: 'center' },
  text: {
    fontSize: 16,
    fontWeight: config.fontWeights.FONT_WEIGHT_MEDIUM,
    marginHorizontal: 20,
    marginVertical: 10,
  },
  imageWrapper: {
    borderRadius: 5,
    shadowColor: config.colors.black,
    shadowOffset: { width: 10, height: 10 },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 7,
  },
  image: {
    width: config.imageWidth,
    height: config.imageHeight,
    resizeMode: 'contain',
    borderRadius: 5,
    marginBottom: 30,
  },
});

export default ImageCard;
