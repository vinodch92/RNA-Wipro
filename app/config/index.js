import colors from './colors';
import localConfig from './localConfig';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  fontWeights,
  accessibilityStrings,
};

export default config;
