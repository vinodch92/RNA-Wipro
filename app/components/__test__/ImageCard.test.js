import React from 'react';
import ImageCard from '../ImageCard';
import { render } from '@testing-library/react-native';
import config from '../../config';

describe('ImageCard', () => {
  const props = {
    containerStyle: { backgroundColor: config.colors.appTheme },
    textStyle: { color: config.colors.white },
    textNumLines: 1,
    imageWrapperStyle: { shadowColor: config.colors.black },
  };

  const authorName = 'Paul Jarvis';
  const id = 10;

  it('matches snapshot', () => {
    expect(render(<ImageCard {...props} />)).toMatchSnapshot();
    expect(render(<ImageCard {...props} authorName={authorName} id={id} />)).toMatchSnapshot();
  });

  it('renders the author name', () => {
    const { queryByTestId } = render(<ImageCard {...props} authorName={authorName} id={id} />);
    expect(queryByTestId(config.accessibilityStrings.AUTHOR_NAME_TEXT)).toBeTruthy();
  });

  it('does not render the author name', () => {
    const { queryByTestId } = render(<ImageCard {...props} />);
    expect(queryByTestId(config.accessibilityStrings.AUTHOR_NAME_TEXT)).toBeFalsy();
  });
});
