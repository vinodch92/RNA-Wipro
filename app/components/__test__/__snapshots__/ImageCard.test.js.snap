// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "backgroundColor": "#34424A",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 5,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 10,
        },
        Object {
          "shadowColor": "#000000",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/200/300?image=0",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "height": 300,
          "marginBottom": 30,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
  </View>
</View>
`;

exports[`ImageCard matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "backgroundColor": "#34424A",
      },
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "500",
          "marginHorizontal": 20,
          "marginVertical": 10,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
    testID="author-name-text"
  >
    Paul Jarvis
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 5,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 10,
        },
        Object {
          "shadowColor": "#000000",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/200/300?image=10",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "height": 300,
          "marginBottom": 30,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
